File: frontend/src/types/instagram.ts
export type AccountStatus = 'active' | 'expired' | 'revoked';

export interface InstagramAccount {
  id: number;
  instagram_user_id: string;
  username: string;
  profile_picture: string | null;
  account_type: string | null;  // "Personal", "Creator", "Business"
  follower_count: number | null;
  status: AccountStatus;
  created_at: string;
}

File: backend/app/schemas/instagram.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class InstagramAccountResponse(BaseModel):
    id: int
    instagram_user_id: str
    username: str
    profile_picture: Optional[str] = None
    account_type: Optional[str] = None
    follower_count: Optional[int] = None
    status: str  # "active", "expired", "revoked"
    created_at: datetime

    class Config:
        from_attributes = True


class InstagramAccountCreate(BaseModel):
    instagram_user_id: str
    username: str
    profile_picture: Optional[str] = None
    account_type: Optional[str] = None
    follower_count: Optional[int] = None
